src/main/java/com/zontik/groshiky/security/AppAuthenticationProvider.java

```diff
@@ -0,0 +1,53 @@
+package com.zontik.groshiky.security;
+
+import com.zontik.groshiky.dao.IRoleDao;
+import com.zontik.groshiky.dao.IUserDao;
+import com.zontik.groshiky.model.Role;
+import com.zontik.groshiky.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+import sun.security.util.Password;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class AppAuthenticationProvider implements AuthenticationProvider {
+
+    @Autowired
+    private IUserDao userDao;
+    @Autowired
+    private IRoleDao roleDao;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+
+    @Override
+    public Authentication authenticate(Authentication auth) throws AuthenticationException {
+        User user = userDao.findByLogin(auth.getName());
+        if(user == null) {
+            throw new UsernameNotFoundException(String.format("User not found: %s", auth.getName()));
+        }
+        String password = auth.getCredentials().toString();
+
+        if(!passwordEncoder.matches(password, user.getPassword())){
```

Comment:
Sorry, perhaps I miss something, but I don't see such a method in the User class. Where it comes from?

Reply:
I use lombok library for getter and setter generation